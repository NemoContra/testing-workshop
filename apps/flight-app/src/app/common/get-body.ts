import { inject } from '@angular/core';
import { DOCUMENT } from '@angular/common';

export const getHTMLBodyElement = (): HTMLBodyElement =>
  inject(DOCUMENT).body as HTMLBodyElement;
