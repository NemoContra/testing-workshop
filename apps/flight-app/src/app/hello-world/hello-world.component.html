<p>Hello {{ name }}! <span *ngIf="!isFine">Geh doch nach Hause!</span></p>
<p data-tid="fine">I am fine: {{ isFine | yesNo }}</p>
