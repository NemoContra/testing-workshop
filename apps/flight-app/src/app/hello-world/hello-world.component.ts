import { Component, Input } from '@angular/core';

@Component({
  selector: 'flight-hello-world',
  templateUrl: './hello-world.component.html',
  styleUrls: ['./hello-world.component.scss'],
})
export class HelloWorldComponent {
  @Input() name = 'world';
  @Input() isFine = true;

  arr = [1, 2, 3];
}
