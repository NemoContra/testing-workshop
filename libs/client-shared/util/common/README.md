# client-shared-util-common

This library was generated with [Nx](https://nx.dev).

## Running unit tests

Run `nx test client-shared-util-common` to execute the unit tests.
