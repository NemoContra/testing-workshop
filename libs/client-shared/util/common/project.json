{
  "projectType": "library",
  "sourceRoot": "libs/client-shared/util/common/src",
  "prefix": "testing-workshop",
  "targets": {
    "test": {
      "executor": "@nrwl/jest:jest",
      "outputs": ["coverage/libs/client-shared/util/common"],
      "options": {
        "jestConfig": "libs/client-shared/util/common/jest.config.ts",
        "passWithNoTests": true
      }
    },
    "lint": {
      "executor": "@nrwl/linter:eslint",
      "options": {
        "lintFilePatterns": [
          "libs/client-shared/util/common/**/*.ts",
          "libs/client-shared/util/common/**/*.html"
        ]
      }
    }
  },
  "tags": ["scope:client-shared", "type:util"]
}
