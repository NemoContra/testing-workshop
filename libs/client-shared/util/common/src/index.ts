export * from './lib/pipes/yes-no.pipe';
export * from './lib/helpers/get-query-params';
