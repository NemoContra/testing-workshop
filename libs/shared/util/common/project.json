{
  "$schema": "../../../../node_modules/nx/schemas/project-schema.json",
  "sourceRoot": "libs/shared/util/common/src",
  "projectType": "library",
  "targets": {
    "lint": {
      "executor": "@nrwl/linter:eslint",
      "outputs": ["{options.outputFile}"],
      "options": {
        "lintFilePatterns": ["libs/shared/util/common/**/*.ts"]
      }
    },
    "test": {
      "executor": "@nrwl/jest:jest",
      "outputs": ["{workspaceRoot}/coverage/libs/shared/util/common"],
      "options": {
        "jestConfig": "libs/shared/util/common/jest.config.ts",
        "passWithNoTests": true
      }
    }
  },
  "tags": ["scope:shared", "type:util"]
}
