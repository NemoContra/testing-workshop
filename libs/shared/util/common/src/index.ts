export * from './lib/helpers/create-random-id/create-random-id';
export * from './lib/rxjs-operators/skip-falsy';
