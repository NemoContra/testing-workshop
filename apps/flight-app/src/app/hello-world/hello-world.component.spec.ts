import { HelloWorldComponent } from './hello-world.component';
import { createHostFactory } from '@ngneat/spectator/jest';
import { YesNoPipe } from '@testing-workshop/client-shared/util/common';
import { testIdSelector } from '@testing-workshop/shared/util/test-helpers';

describe('HelloWorldComponent', () => {
  const createComponent = createHostFactory({
    component: HelloWorldComponent,
    imports: [YesNoPipe],
  });

  it('should render "hello world"!', () => {
    const spectator = createComponent(
      `<flight-hello-world></flight-hello-world>`
    );
    expect(spectator.fixture).toMatchSnapshot();
  });

  it('should render "hello Lohr"!', () => {
    const spectator = createComponent(
      `<flight-hello-world name="Lohr"></flight-hello-world>`
    );
    expect(spectator.fixture).toMatchSnapshot();
  });

  it('should not be fine', () => {
    const spectator = createComponent(
      `<flight-hello-world name="Lohr" [isFine]="false"></flight-hello-world>`
    );

    const fine = spectator.query(testIdSelector('fine'));

    expect(fine).toMatchInlineSnapshot(`
      <p
        data-tid="fine"
      >
        I am fine: no
      </p>
    `);
  });
});
