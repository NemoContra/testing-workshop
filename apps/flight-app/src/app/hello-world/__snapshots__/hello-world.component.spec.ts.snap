// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HelloWorldComponent should render "hello Lohr"! 1`] = `
<ng-component
  __ngContext__={[Function Number]}
>
  <flight-hello-world
    name="Lohr"
  >
    <p>
      Hello Lohr! 
    </p>
    <p
      data-tid="fine"
    >
      I am fine: yes
    </p>
  </flight-hello-world>
</ng-component>
`;

exports[`HelloWorldComponent should render "hello world"! 1`] = `
<ng-component
  __ngContext__="0"
>
  <flight-hello-world>
    <p>
      Hello world! 
    </p>
    <p
      data-tid="fine"
    >
      I am fine: yes
    </p>
  </flight-hello-world>
</ng-component>
`;
